Extensions/Physics2Behavior/JsExtension.js

```diff
@@ -0,0 +1,2988 @@
+      behaviorProperties.set(
+        "masks",
+        new gd.PropertyDescriptor(masks)
+      );
+
+      return behaviorProperties;
+    };
+    physics2Behavior.setRawJSONContent(
+      JSON.stringify({
+        type: "Dynamic",
+        bullet: false,
+        fixedRotation: false,
+        canSleep: true,
+        shape: "Box",
+        shapeDimensionA: 0,
+        shapeDimensionB: 0,
+        shapeOffsetX: 0,
+        shapeOffsetY: 0,
+        density: 1.0,
+        friction: 0.3,
+        restitution: 0.1,
+        linearDamping: 0.1,
+        angularDamping: 0.1,
+        gravityScale: 1,
+        layers: 1,
+        masks: 1
+      })
+    );
+
+    var sharedData = new gd.BehaviorSharedDataJsImplementation();
+    sharedData.updateProperty = function(
+      sharedContent,
+      propertyName,
+      newValue
+    ) {
+      if (propertyName === "gravityX") {
+        sharedContent.gravityX = parseInt(newValue, 10);
+        return true;
+      }
+      if (propertyName === "gravityY") {
+        sharedContent.gravityY = parseInt(newValue, 10);
+        return true;
+      }
+      if (propertyName === "scaleX") {
+        newValue = parseInt(newValue, 10);
+        if(newValue <= 0) newValue = 1;
+        sharedContent.scaleX = newValue;
+        return true;
+      }
+      if (propertyName === "scaleY") {
+        newValue = parseInt(newValue, 10);
+        if(newValue <= 0) newValue = 1;
+        sharedContent.scaleY = newValue;
+        return true;
+      }
+
+      return false;
+    };
+    sharedData.getProperties = function(sharedContent) {
+      var sharedProperties = new gd.MapStringPropertyDescriptor();
+
+      sharedProperties.set(
+        "gravityX",
+        new gd.PropertyDescriptor(
+          sharedContent.gravityX.toString(10)
+        ).setType("Number")
+      );
+      sharedProperties.set(
+        "gravityY",
+        new gd.PropertyDescriptor(
+          sharedContent.gravityY.toString(10)
+        ).setType("Number")
+      );
+      sharedProperties.set(
+        "scaleX",
+        new gd.PropertyDescriptor(
+          sharedContent.scaleX.toString(10)
+        ).setType("Number")
+      );
+      sharedProperties.set(
+        "scaleY",
+        new gd.PropertyDescriptor(
+          sharedContent.scaleY.toString(10)
+        ).setType("Number")
+      );
+
+      return sharedProperties;
+    };
+    sharedData.setRawJSONContent(
+      JSON.stringify({
+        gravityX: 0,
+        gravityY: 9.8,
+        scaleX: 100,
+        scaleY: 100
+      })
+    );
+
+    var aut = extension
+    // extension
+      .addBehavior(
+        "Physics2Behavior",
+        t("Physics2 behavior (beta)"),
+        "Physics2Behavior",
+        t("Simulate physics, the successor of the old physics behavior"),
+        "",
+        "res/physics32.png",
+        "Physics2Behavior",
+        physics2Behavior,
+        sharedData
+      )
+      .setIncludeFile("Extensions/Physics2/physics2runtimebehavior.js")
```

Comment:
Path should be `Extensions/Physics2Behavior` for all the include file no? I had to change them to get it to work :)

Reply:
Aha!, nice catch, when I renamed the folder removed the original one everywhere, maybe I've forgotten to remove it in the previews folder, loading the extension  anyway

Reply:
Yeah, I think there was still the folder in the preview folder so you didn't see the issue :)